openlibrary/accounts/__init__.py

```diff
@@ -11,6 +11,35 @@ def get_group(name):
     return web.ctx.site.get("/usergroup/%s"%name)
 
 
+class RunAs(object):
+    """
+    Escalates privileges to become username, performs action as user,
+    and then de-escalates to original user.
+    """
+
+    def __init__(self, username):
+        """
+        :param str username: Username e.g. /people/mekBot of user to run action as
+        """
+        self.tmp_account = find(username=username)
+        self.calling_user_auth_token = None
+
+        if not self.tmp_account:
```

Comment:
@cdrini what if user is not logged in? The escalation should still be performed.

Reply:
Good point; let's drop the error check and only set auth token if the user account exists (inside __enter__). Would that work?

Reply:
Wait nevermind; tmp_account is the user you want to run as, so it has to exist.

Reply:
You're right!